### PR TITLE
Comment rendering

### PR DESCRIPTION
**Pull request by tkknight**
### Description

Relates to https://github.com/SciTools/iris/issues/3840 and the issue https://github.com/SciTools/iris/issues/4657.

The docstrings layout is inconsistent especially the indentations.

* `dot.py` [latest](https://scitools-iris.readthedocs.io/en/latest/generated/api/iris/fileformats/dot.html) | [this PR](https://tkknight-iris-test-doc.readthedocs.io/en/latest/generated/api/iris/fileformats/dot.html)
* `nimrod_load_rules.py` [latest](https://scitools-iris.readthedocs.io/en/latest/generated/api/iris/fileformats/nimrod_load_rules.html) | [this PR](https://tkknight-iris-test-doc.readthedocs.io/en/latest/generated/api/iris/fileformats/nimrod_load_rules.html
)
* `pp.py` [latest](https://scitools-iris.readthedocs.io/en/latest/generated/api/iris/fileformats/pp.html#iris.fileformats.pp.save) | [this PR](https://tkknight-iris-test-doc.readthedocs.io/en/latest/generated/api/iris/fileformats/pp.html#iris.fileformats.pp.save)

Changes made as per the official numpdoc strings guide, see https://numpydoc.readthedocs.io/en/latest/format.html#docstring-standard and https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_numpy.html

I also updated the docstring guidance on the docs and the pull request guidance

* `docstrings.rst` [latest](https://scitools-iris.readthedocs.io/en/latest/developers_guide/documenting/docstrings.html) | [this PR](https://tkknight-iris-test-doc.readthedocs.io/en/latest/developers_guide/documenting/docstrings.html)
* `contributing_pull_request_checklist.rst` [latest](https://scitools-iris.readthedocs.io/en/latest/developers_guide/contributing_pull_request_checklist.html) | [this PR](https://tkknight-iris-test-doc.readthedocs.io/en/latest/developers_guide/contributing_pull_request_checklist.html)

For help with the docstring formatting I used pydocstyle http://www.pydocstyle.org/en/stable/usage.html via the command:

```
pip install pydocstyle
pydocstyle --convention=numpy pp.py --add-ignore=D200,D405,D407,D411,D205,D400,D105,D102,D401 # D205 D400 D105 D102 D401
```

I chose to ignore some for now.  We have an option to make this part of our pre-commit checks in future.

---
[Consult Iris pull request check list]( https://scitools-iris.readthedocs.io/en/latest/developers_guide/contributing_pull_request_checklist.html)

### CODE REVIEW

File: lib/iris/fileformats/nimrod_load_rules.py [hunk 1]
```diff
@@ -905,17 +918,18 @@ def run(field, handle_metadata_errors=True):
...
-
-        * field - a :class:`~iris.fileformats.nimrod.NimrodField`
-
-        * handle_metadata_errors - Set to False to omit handling of known meta-data deficiencies
-                                   in Nimrod-format data
-
-    Returns:
+    Args
+    ----
+    field: :class:`~iris.fileformats.nimrod.NimrodField`
```
Comment (wjbenfold) [inline lib/iris/fileformats/nimrod_load_rules.py:923]: Not sure why in preview this has a "-" on the end of the line?

Comment (wjbenfold) [reply lib/iris/fileformats/nimrod_load_rules.py:923]: Happens a few times - I think it's expecting you to follow the line with a description? Looks particularly odd where it happens after `**kwargs`

**Review by wjbenfold — CHANGES_REQUESTED** [review]
Couple of comments but it looks nice. Probably worth describing the main differences that you've identified to the team somewhere/somewhen?

File: lib/iris/fileformats/pp.py [hunk 2]
```diff
@@ -2062,24 +2060,23 @@ def load_cubes(filenames, callback=None, constraints=None):
...
-        The resultant cubes may not be in the order that they are in the file
-        (order is not preserved when there is a field with orography
-        references)
+    Load cubes from a list of pp filenames containing little-endian data.
+
+    Args
+    ----
+    filenames
+        list of pp filenames to load
+    Kwargs
```
[the remark this answers is not in the document]
Comment (tkknight) [reply lib/iris/fileformats/pp.py:2069]: I chose to use `**kwargs` as per this official guidance https://numpydoc.readthedocs.io/en/latest/format.html#parameters, but did not change this one - will update.

File: lib/iris/fileformats/nimrod_load_rules.py (hunk 1 above)
Comment (tkknight) [reply lib/iris/fileformats/nimrod_load_rules.py:923]: This is expected behaviour I believe.  When defining the Args you can:

* define the return type
* define the description of the return
* do both

The one you queried just does the first, hence the hyphen when the description would be but there is none defined.  For me, this PR was about fixing the formatting and doing some light improvements, someone with more familiarity with Iris will need to review and improve the quality of the docstrings generally.

File: lib/iris/fileformats/pp.py [hunk 3]
```diff
@@ -2192,26 +2193,30 @@ def save(cube, target, append=False, field_coords=None):
     """
     Use the PP saving rules (and any user rules) to save a cube to a PP file.
 
-    Args:
+    Args
+    ----
+    cube: :class:`~iris.fileformats.nimrod.NimrodField`
```
[the remark this answers is not in the document]
Comment (tkknight) [reply lib/iris/fileformats/pp.py:2198]: You are correct.  Copy and paste gremlim here.... will fix.

**Review by wjbenfold — APPROVED** [review]

**Comment by wjbenfold** [comment]
We're getting 403 errors from GitHub that are breaking the linkcheck, and it doesn't seem to resolve on a rerun?

**Pull request merged**